pyfifotax/report_data.py

```diff
@@ -0,0 +1,509 @@
+                    date=event.date,
+                    value=event.bonus,
+                )
+
+                bought_shares = awv.AWVEntryZ10ESPPDeposit(
+                    date=event.date,
+                    symbol=event.symbol,
+                    currency=event.currency,
+                    quantity=event.received_shares.quantity,
+                    value=event.received_shares.total_buy_value(),
+                )
+
+                self.awv_z4_events.append(bonus)
+                self.awv_z10_events.append(bought_shares)
+
+            elif isinstance(event, DividendEvent):
+                self.held_forex[event.currency].push(event.received_net_dividend)
+
+                if event.withheld_tax is not None:
+                    self.misc["Tax Withholding"].append(event.withheld_tax)
+                self.misc["Dividend Payments"].append(event.received_dividend)
+
+                # dividends should be small enough to not trigger AWV reportings
+
+            elif isinstance(event, BuyEvent):
+                # if not enough money, pop on FOREX Queue will fail
+                tmp = self.held_forex[event.currency].pop(
+                    event.cost_of_shares,
+                    to_decimal(1),
+                    event.date,
+                )
+                self.sold_forex[event.currency].extend(tmp)
+
+                self.held_shares[event.symbol].push(event.received_shares)
+                if event.paid_fees is not None:
+                    self.misc["Fees"].append(event.paid_fees)
+
+                self.awv_z10_events.append(
+                    awv.AWVEntryZ10Buy(
+                        date=event.date,
+                        symbol=event.received_shares.symbol,
+                        currency=event.currency,
+                        quantity=event.received_shares.quantity,
+                        value=event.received_shares.total_buy_value(),
+                    )
+                )
+
+            elif isinstance(event, SellEvent):
+                # if not enough shares to sell, pop on SHARE Queue will fail
+                tmp = self.held_shares[event.symbol].pop(
+                    event.quantity, event.sell_price, event.date
+                )
+                self.sold_shares[event.symbol].extend(tmp)
+
+                self.held_forex[event.currency].push(event.received_forex)
+
+                if event.paid_fees is not None:
+                    self.misc["Fees"].append(event.paid_fees)
+
+                self.awv_z10_events.append(
+                    awv.AWVEntryZ10Sale(
+                        date=event.date,
+                        symbol=event.symbol,
+                        currency=event.currency,
+                        quantity=event.quantity,
+                        value=event.quantity * event.sell_price,
+                    )
+                )
+
+            elif isinstance(event, MoneyTransferEvent):
+                if event.amount > 0:
```

Comment:
In a way, it is the wrong date, yes. The thing is, for EUR, `buy_date` actually doesn't matter. As `buy_date` could be empty, however, I have the following snippet when reading things
```        
buy_date = (
            datetime(1, 1, 1)
            if row.currency == "EUR" or row.amount < 0
            else row.buy_date
        )
```
This is really just a dummy to not break typing and other things. Although, not the best. "fees" on the other hand, currently doesn't have a dedicated handling. Thats hould be set to 0 to avoid issues.